src/tracking/lib/amrit-tracking.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Inject } from '@angular/core';
+import { TrackingProvider } from './tracking-provider';
+import { SessionStorageService } from '../../registrar/services/session-storage.service';
+import { Router, NavigationEnd } from '@angular/router';
+import { filter } from 'rxjs/operators';
+import { TRACKING_PROVIDER } from './tracking.tokens';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AmritTrackingService {
+  constructor(
+    @Inject(TRACKING_PROVIDER) private trackingProvider: TrackingProvider,
+    private sessionStorage: SessionStorageService,
+    private router: Router
+  ) {
+    // Initialise underlying provider
+    this.trackingProvider.init?.();
+    
+    // Initialize automatic page view tracking
+    this.setupPageViewTracking();
+    
+    // Set user ID if available
+    this.setupUserTracking();
+  }
+
+  private setupPageViewTracking() {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd)
+    ).subscribe((event: NavigationEnd) => {
+      this.trackingProvider.pageView(event.urlAfterRedirects);
+    });
+  }
+
+  private setupUserTracking() {
+    const userId = this.sessionStorage.getItem('userID');
```

Comment:
please type-check the userId and confirm if its not empty string value as well to avoid runtime errors